src/main/java/org/opensearch/knn/plugin/transport/KNNProfileResponse.java

```diff
@@ -0,0 +1,260 @@
+ * at segment, shard, and cluster levels.
+ *
+ * Example response:
+ * {
+ *   "total_shards": 2,
+ *   "successful_shards": 2,
+ *   "failed_shards": 0,
+ *   "shard_profiles": {
+ *     "0": {
+ *       "segments": [
+ *         {
+ *           "segment_id": "_0",
+ *           "dimension": 128,
+ *           "vector_statistics": [
+ *             {
+ *               "dimension_index": 0,
+ *               "statistics": {
+ *                 "count": 1000,
+ *                 "min": -0.523,
+ *                 "max": 0.785,
+ *                 "sum": 156.78,
+ *                 "mean": 0.157,
+ *                 "geometric_mean": 0.145,
+ *                 "variance": 0.089,
+ *                 "std_deviation": 0.298,
+ *                 "sum_of_squares": 245.67
+ *               }
+ *             }
+ *             // ... additional dimensions
+ *           ]
+ *         }
+ *         // ... additional segments
+ *       ],
+ *       "aggregated": {
+ *         "total_segments": 3,
+ *         "dimension": 128,
+ *         "dimensions": [
+ *           {
+ *             "dimension_id": 0,
+ *             "count": 3000,
+ *             "min": -0.723,
+ *             "max": 0.892,
+ *             "mean": 0.167,
+ *             "std_deviation": 0.312,
+ *             "sum": 501.23,
+ *             "variance": 0.097
+ *           }
+ *           // ... additional dimensions
+ *         ]
+ *       }
+ *     }
+ *     // ... additional shards
+ *   },
+ *   "cluster_aggregation": {
+ *     "total_shards": 2,
+ *     "dimension": 128,
+ *     "dimensions": [
+ *       {
+ *         "dimension_id": 0,
+ *         "count": 6000,
+ *         "min": -0.723,
+ *         "max": 0.892,
+ *         "mean": 0.172,
+ *         "std_deviation": 0.315,
+ *         "sum": 1032.45,
+ *         "variance": 0.099
+ *       }
+ *       // ... additional dimensions
+ *     ]
+ *   },
+ *   "failures": []
+ * }
+ */
+public class KNNProfileResponse extends BroadcastResponse implements ToXContentObject {
+
+    List<KNNIndexShardProfileResult> shardProfileResults;
+
+    public KNNProfileResponse() {}
+
+    public KNNProfileResponse(StreamInput in) throws IOException {
+        super(in);
+    }
+
+    public KNNProfileResponse(
+        List<KNNIndexShardProfileResult> shardProfileResults,
+        int totalShards,
+        int successfulShards,
+        int failedShards,
+        List<DefaultShardOperationFailedException> shardFailures
+    ) {
+        super(totalShards, successfulShards, failedShards, shardFailures);
+
+        this.shardProfileResults = shardProfileResults;
+    }
+
+    @Override
+    public void writeTo(StreamOutput streamOutput) throws IOException {
+        throw new UnsupportedOperationException("This method is not available");
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
I dont think we need to implement this method. See https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/support/broadcast/BroadcastResponse.java#L171-L183. We should just implement https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/support/broadcast/BroadcastResponse.java#L183.